package/tests/test_integrations/test_sqlite_store.py

```diff
@@ -67,9 +216,138 @@ def test_update_git_branch(self, store_path, mocker):
+        mock_log = mocker.patch.object(logging.Logger, "exception")
+        sqlite_store._upload()
+        mock_log.assert_called_once_with("Upload failed: Connection error")
+
+    def test_download_from_s3_success(
+        self, mocker, store_path, remote_path, mocked_db_in_s3, tmp_path
+    ):
+        mocker.patch("fsspec.filesystem")
+        sqlite_store = SQLiteStore(
+            store_path, next(session_id()), remote_path=remote_path
+        )
+        sqlite_store._remote_fs.glob.return_value = mocked_db_in_s3
+        sqlite_store._download()
+        downloaded_dbs = set(Path(sqlite_store.location).parent.glob("*.db")) - {
+            Path(sqlite_store.location)
+        }
+        assert downloaded_dbs == {Path(tmp_path / "db1.db"), Path(tmp_path / "db2.db")}
+
+    def test_download_from_s3_failure(self, mocker, store_path, remote_path):
+        mocker.patch("fsspec.filesystem")
+        sqlite_store = SQLiteStore(
+            store_path, next(session_id()), remote_path=remote_path
+        )
+        sqlite_store._remote_fs.glob.side_effect = ConnectionError("Connection error")
+        mock_log = mocker.patch.object(logging.Logger, "exception")
+        sqlite_store._download()
+        downloaded_dbs = set(Path(sqlite_store.location).parent.glob("*.db")) - {
+            Path(sqlite_store.location)
+        }
+        # Assert that the number of databases downloaded is 0
+        assert len(downloaded_dbs) == 0
+        mock_log.assert_called_once_with("Download failed: Connection error")
+
+    def test_merge_databases(
+        self,
+        mocker,
+        store_path,
+        remote_path,
+        mocked_db_in_s3,
+    ):
+        mocker.patch("fsspec.filesystem")
+        sqlite_store = SQLiteStore(
+            store_path, next(session_id()), remote_path=remote_path
+        )
+        sqlite_store._remote_fs.glob.return_value = mocked_db_in_s3
+        sqlite_store._download()
+        sqlite_store._merge()
+        db_session = sqlite_store._db_session_class
+        with db_session() as session:
+            assert session.query(RunModel).count() == 2
+
+    def test_merge_databases_with_repeated_runs(
+        self,
+        mocker,
+        store_path,
+        remote_path,
+        mocked_db_in_s3_repeated_runs,
+    ):
+        mocker.patch("fsspec.filesystem")
+        sqlite_store = SQLiteStore(
+            store_path, next(session_id()), remote_path=remote_path
+        )
+        sqlite_store._remote_fs.glob.return_value = mocked_db_in_s3_repeated_runs
+        sqlite_store._download()
+        sqlite_store._merge()
+        db_session = sqlite_store._db_session_class
+        downloaded_dbs = set(Path(sqlite_store.location).parent.glob("*.db")) - {
+            Path(sqlite_store.location)
+        }
+        assert len(downloaded_dbs) == 3
+        with db_session() as session:
+            assert session.query(RunModel).count() == 2
```

Comment:
Same suggestion again here about asserting you have the right run ids.

Reply:
done